x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/process/AnalyticsResultProcessor.java

```diff
@@ -231,9 +235,37 @@ private TrainedModelConfig createTrainedModelConfig(TrainedModelDefinition.Build
             .setInput(new TrainedModelInput(fieldNamesWithoutDependentVariable))
             .setLicenseLevel(License.OperationMode.PLATINUM.description())
             .setDefaultFieldMap(defaultFieldMapping)
+            .setInferenceConfig(buildInferenceConfig(definition.getTrainedModel().targetType()))
             .build();
     }
 
+    private InferenceConfig buildInferenceConfig(TargetType targetType) {
+        switch (targetType) {
+            case CLASSIFICATION:
+                assert analytics.getAnalysis() instanceof Classification;
+                Classification classification = ((Classification)analytics.getAnalysis());
+                return new ClassificationConfig(classification.getNumTopClasses(),
+                    ClassificationConfig.DEFAULT_RESULTS_FIELD,
+                    ClassificationConfig.DEFAULT_TOP_CLASSES_RESULTS_FIELD,
+                    classification.getBoostedTreeParams().getNumTopFeatureImportanceValues() == null ?
+                        0 :
+                        classification.getBoostedTreeParams().getNumTopFeatureImportanceValues());
+            case REGRESSION:
+                assert analytics.getAnalysis() instanceof Regression;
+                Regression regression = ((Regression)analytics.getAnalysis());
+                return new RegressionConfig(RegressionConfig.DEFAULT_RESULTS_FIELD,
+                    regression.getBoostedTreeParams().getNumTopFeatureImportanceValues() == null ?
+                        0 :
```

Comment:
Let `RegressionConfig` pick the default value (it does this in the constructor anyway)

```
RegressionConfig(RegressionConfig.DEFAULT_RESULTS_FIELD,
                    regression.getBoostedTreeParams().getNumTopFeatureImportanceValues())
```

Reply:
Same with `ClassificationConfig`